src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -255,101 +266,117 @@ export default function parseAdaptationSets(
         }
 
         const isClosedCaption = type === "text" &&
-                                accessibility != null &&
+                                accessibility !== undefined &&
                                 isHardOfHearing(accessibility) ? true :
                                                                  undefined;
         const isAudioDescription = type === "audio" &&
-                                   accessibility != null &&
+                                   accessibility !== undefined &&
                                    isVisuallyImpaired(accessibility) ? true :
                                                                        undefined;
+
+        const isTrickModeTrack = trickModeAttachedAdaptationIds !== undefined;
         const adaptationID = newID = getAdaptationID(adaptation,
                                                      representations,
                                                      { isClosedCaption,
                                                        isAudioDescription,
-                                                       type });
+                                                       type },
+                                                       isTrickModeTrack);
         const parsedAdaptationSet : IParsedAdaptation = { id: adaptationID,
                                                           representations,
                                                           type };
-        if (adaptation.attributes.language != null) {
+        if (adaptation.attributes.language !== undefined) {
           parsedAdaptationSet.language = adaptation.attributes.language;
         }
-        if (isClosedCaption != null) {
+        if (isClosedCaption !== undefined) {
           parsedAdaptationSet.closedCaption = isClosedCaption;
         }
-        if (isAudioDescription != null) {
+        if (isAudioDescription !== undefined) {
           parsedAdaptationSet.audioDescription = isAudioDescription;
         }
         if (isDub === true) {
           parsedAdaptationSet.isDub = true;
         }
 
-        const adaptationsOfTheSameType = parsedAdaptations[type];
-        if (adaptationsOfTheSameType === undefined) {
-          parsedAdaptations[type] = [parsedAdaptationSet];
-          if (isMainAdaptation && type === "video") {
-            acc.videoMainAdaptation = parsedAdaptationSet;
-          }
+        if (trickModeAttachedAdaptationIds !== undefined) {
+          acc.trickModeAdaptations.push({ adaptation: parsedAdaptationSet,
+                                          trickModeAttachedAdaptationIds });
         } else {
-          let mergedInto : IParsedAdaptation|null = null;
+          const adaptationsOfTheSameType = parsedAdaptations[type];
```

Comment:
This becomes too hard to read.

Can't we return early when having a trickmode track?